src/subcommand/server.rs

```diff
@@ -2225,6 +2234,63 @@ impl Server {
     })
   }
 
+  async fn parent_inscriptions_recursive(
+    Extension(index): Extension<Arc<Index>>,
+    Path(inscription_id): Path<InscriptionId>,
+  ) -> ServerResult {
+    Self::parent_inscriptions_recursive_paginated(Extension(index), Path((inscription_id, 0))).await
+  }
+
+  async fn parent_inscriptions_recursive_paginated(
+    Extension(index): Extension<Arc<Index>>,
+    Path((child, page)): Path<(InscriptionId, usize)>,
+  ) -> ServerResult {
+    task::block_in_place(|| {
+      let entry = index
+        .get_inscription_entry(child)?
+        .ok_or_not_found(|| format!("inscription {child}"))?;
+
+      let (ids, more) = index.get_parents_by_sequence_number_paginated(entry.parents, 100, page)?;
+
+      let parents = ids
+        .into_iter()
+        .map(|inscription_id| {
+          let entry = index
+            .get_inscription_entry(inscription_id)
+            .unwrap()
```

Comment:
These errors should be handled. If the result type of a map is `Result<T>`, then you can collect into a `Result<Vec<T>>`, which will return an error if any of the results is `Result::Err`.